enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
@rhuss I never said we should silently ignore the second init container. My point is that when these errors crop up now and again due to refactorings of fmp and because we don't yet have sufficient CI across lots of independently released projects - I prefer to focus on generating valid kubernetes resources first and generating warnings instead of breaking things.

Maybe we could have a 'strict' mode that always fails the build instead of generating warnings so folks can turn on strict mode in CI builds to ensure that CI jobs fail if duplicates are generated - but avoid breaking release pipelines?